apm-sniffer/optional-plugins/optional-spring-plugins/spring-annotation-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/annotations/services/SpringServicesInstrumentation.java

```diff
@@ -19,16 +19,21 @@
 package org.apache.skywalking.apm.plugin.spring.annotations.services;
 
 import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+import org.apache.skywalking.apm.agent.core.plugin.match.logical.LogicalMatchOperation;
 import org.apache.skywalking.apm.plugin.spring.annotations.AbstractSpringBeanInstrumentation;
 
 import static org.apache.skywalking.apm.agent.core.plugin.match.ClassAnnotationMatch.byClassAnnotationMatch;
+import static org.apache.skywalking.apm.agent.core.plugin.match.RegexMatch.byRegexMatch;
 
 public class SpringServicesInstrumentation extends AbstractSpringBeanInstrumentation {
 
     public static final String ENHANCE_ANNOTATION = "org.springframework.stereotype.Service";
 
     @Override
     protected ClassMatch enhanceClass() {
-        return byClassAnnotationMatch(new String[] {ENHANCE_ANNOTATION});
+        return LogicalMatchOperation.and(
```

Comment:
Should only activate `and byRegexMatch` if the  `#getRegexExpressions` return not null. Then, it is better for matching mechanism performance.

Reply:
resolved